File: linked_list/lib/linked_list.rb
require "./lib/node"

class LinkedList
  def initialize
    @head = nil
  end

  def push(data)
    new_node = Node.new(data)
    if @head == nil
      @head = new_node
    else
      @head.push(new_node)
    end
  end
end

File: linked_list/src/linked_list.js
var Node = require('./node.js')

class LinkedList {
  constructor() {
    this.head = null
  }

  push(data) {
    var newNode = new Node(data)
    if (this.head === null) {
      this.head = newNode
    } else {
      this.head.push(newNode)
    }
  }
}

module.exports = LinkedList
